tests/components/wled/test_light.py

```diff
@@ -587,3 +588,46 @@ async def test_effect_service_error(
         state = hass.states.get("light.wled_rgb_light_segment_0")
         assert state.state == STATE_ON
         assert "Invalid response from API" in caplog.text
+
+
+async def test_preset_service(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the prest service of a WLED light."""
+    await init_integration(hass, aioclient_mock)
+
+    with patch("wled.WLED.preset") as light_mock:
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_PRESET,
+            {
+                ATTR_ENTITY_ID: "light.wled_rgb_light_segment_0",
+                ATTR_PRESET: 1,
+            },
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        light_mock.assert_called_once_with(
+            preset=1,
+        )
+
+
```

Comment:
Is there a scenario where a preset may return an error (ie. if a user has not configured that preset on their system)? I don't know the specific capabilities of the WLED, but in other systems I may have 16 available presets, but I may only configure 4. What is the expected result from the API if you call for preset #5 and it isn't present? If that case exists there should be a test case for this as well.

Reply:
The API will return an error that is handled by the `@wled_exception_handler` decorator.

Reply:
I covered this a little bit in `light.py` but no error is thrown if a non-existent preset is called. The JSON API returns a value of `success` and the `ps` value simply changes to `-1` (no preset active).

Reply:
Depends a bit on the version of WLED on the behavior, however, HA is able to recover from either case.